modules/nixos/sbd-server.nix

```diff
@@ -0,0 +1,124 @@
+{
+  self,
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  cfg = config.services.sbd-server;
+  types = lib.types;
+in {
+  options.services.sbd-server = {
+    enable = lib.mkEnableOption "sbd-server";
+
+    package = lib.mkOption {
+      default = self.packages.${pkgs.system}.sbd-serverd;
+      type = types.package;
+    };
+
+    address = lib.mkOption {
+      description = "address to bind";
+      type = types.str;
+    };
+
+    tls-port = lib.mkOption {
+      description = "port to bind for incoming TLS connections";
+      type = types.int;
+    };
+
+    url = lib.mkOption {
+      description = "url for incoming TLS connections to the signal server";
+      type = types.str;
+    };
+  };
+
+  config = lib.mkIf (cfg.enable) {
+    # TODO: can be tested with check-services tool on the sbd integration branch
+
+    systemd.services.sbd-server = {
+      after = ["network.target"];
+      wantedBy = ["multi-user.target"];
+
+      environment = {
+        TMPDIR = "%T";
+      };
+
+      serviceConfig = {
+        DynamicUser = true;
+        PrivateTmp = true;
+
+        # use this mechanism to let systemd take care of file permissions for the dynamic user it creates
+        LoadCredential = [
+          "cert.pem:${config.security.acme.certs."${cfg.url}".directory}/cert.pem"
+          "key.pem:${config.security.acme.certs."${cfg.url}".directory}/key.pem"
+        ];
+        Restart = "always";
+
+        AmbientCapabilities =
+          # needed for binding to ports <1024
+          lib.lists.optionals (cfg.tls-port
+            < 1024) [
+            "CAP_NET_BIND_SERVICE"
+          ];
+
+        ExecStart = builtins.concatStringsSep " " [
+          (lib.meta.getExe cfg.package)
+
+          # bind to the public interface
+          "--bind=${cfg.address}:${builtins.toString cfg.tls-port}"
+
+          # configure TLS certificates
+          ''--cert-pem-file="''${CREDENTIALS_DIRECTORY}/cert.pem"''
+          ''--priv-key-pem-file="''${CREDENTIALS_DIRECTORY}/key.pem"''
+        ];
+      };
+    };
+
+    networking.firewall.allowedTCPPorts = [
+      80
+      444
+
+      cfg.tls-port
+    ];
+
+    services.nginx = {
+      enable = true;
+      virtualHosts."${cfg.url}" = {
+        serverName = cfg.url;
+        enableACME = true;
+        addSSL = true;
+
+        locations."/".root = "/var/www/${cfg.url}";
+
+        listen = [
+          {
+            addr = "${cfg.address}";
+            port = 80;
+            ssl = false;
+          }
+
+          {
+            addr = "${cfg.address}";
+            port = 444;
+            ssl = true;
+          }
+        ];
+      };
+    };
+
+    security.acme = {
+      acceptTerms = true;
+      defaults = {
+        email = "acme@holo.host";
+      };
+
+      # note: the directory watching tls reload story has not yet been implemented. when tls certs are updated, the service must be restarted
+      certs."${cfg.url}" = {
+        reloadServices = ["sbd-server"];
```

Comment:
FYI @neonphog this causes a `systemctl reload-or-restart` when the certificate changes. we don't have a `ExecReload` specified for the service and thus it'll restart and thus interrupt the service. is it worth putting in a reload signal e.g. `SIGHUP1` instead of implementing the directory watcher in `sbd-serverd`?

Reply:
rust only supports [ctrl-c](https://docs.rs/tokio/1.37.0/tokio/signal/index.html) cross-platform... I'd rather not have os specific code if we can avoid it

Reply:
alright; my final idea is a local-only control socket with a simple integer based protocol. i'm also fine deferring to https://github.com/holochain/sbd/issues/13. please decide and resolve this thread accordingly.

Reply:
I'd prefer to implement 13, it should only be half a day for me, if you're uncomfortable with the servers as-is, we can prioritize it.